bin/grouper-api

```diff
@@ -29,6 +31,66 @@ class Application(tornado.web.Application):
         super(Application, self).__init__(*args, **kwargs)
 
 
+class SendEmailThread(Thread):
+    """Background thread for sending emails from the async pool
+
+    This class thread will exist on multiple servers in a standard Grouper production environment
+    so we need to ensure that it's race-safe.
+    """
+
+    def send_emails(self, session, now_ts):
```

Comment:
Do everything apart from actually sending email:  write to db, return number of would-be sent emails, even build the email to send but not send it.